### PR TITLE
Experiment 8A report

### PR DESCRIPTION
**Pull request by OmarTariq612**

### CODE REVIEW

**Comment by AbdulrhmnGhanem** [comment]
Could you add the WER numbers for the oversampling case?

**Comment by AbdulrhmnGhanem** [comment]
Please add the names of the hard examples, and the code used for picking them.

**Comment by OmarTariq612** [comment]
I'm not sure I understood that correctly but here you are:
* a [notebook](https://colab.research.google.com/drive/1JtKyiKnXlysP5bKaX8cgkl756ZXfcNmw?usp=sharing) that represents the hard examples:
* a [notebook](https://colab.research.google.com/drive/1JtKyiKnXlysP5bKaX8cgkl756ZXfcNmw?usp=sharing) that represents the raw WER numbers of the oversampling technique.

Do you mean adding those tables to the report itself ?

**Review by abdoei — APPROVED** [review]

**Comment by AbdulrhmnGhanem** [comment]
Add this to the report

before_vs_after
Also add the total score of the epoch, like what you used to do

https://github.com/ElGarash/Al-Ajwad/blob/24778969db9bd5f1a8fa1f5f0228fd87f2b9ea3b/experiments/04A.md?plain=1#L69-L80

File: experiments/08A.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
+# Experiment 8: To Oversample or Not To Oversample
+
+Auther: Omar Tarek
```
Comment (AbdulrhmnGhanem) [inline experiments/08A.md:3]:
```suggestion
_Sat May 25 07:09:04 PM EEST 2024_

Author: Omar Tarek

Signed-off: Abdulrhmn Ghanem
```

File: experiments/08A.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+* Outlier is any sorah that has a WER greater than the upper bound of **an accetable range** that is calculated using this formula:
+
+        IQR = Q3 - Q1
+        Accebtable range = [Q1 - 1.5 * IQR, Q3 + 1.5 * IQR]
+
+
+    (any sorah that has WER above Q3 + 1.5 * IQR)
+
+* Only sorahs **from 77 to 114** have been considered (not all hard examples). So whenever hard examples are mentioned in this document it means hard examples **from 77 to 114 only**.
```
Comment (AbdulrhmnGhanem) [inline experiments/08A.md:31]:
```suggestion
* Only sorahs **from 77 to 114** have been considered (not all hard examples). So whenever hard examples are mentioned in this document it means hard examples **from 77 to 114 only**. In this experiment the hard examples are:

- النازعات
- عبس
- التكوير
- الإنفطار
- الإنشقاق
- البروج
- الطارق
- الأعلى
- الغاشية
- الفجر
- البلد
- الشمس
- الضحى
- الشرح
- العلق
- البينة
- الزلزلة
- القارعة
- التكاثر
- العصر
- الهمزة
- الفيل
- قريش
- الماعون
- الكوثر
- الكافرون
- النصر
- المسد
- الإخلاص
- الفلق
- الناس
```

File: experiments/08A.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+<p align="center">
+    <img src="./media/08A3.png" alt="Visualization of oversampling the hard examples compared to the normal training strategy">
+</p>
+
+<p align="center">
+    <span style="color: #888888;"><em>Figure 08A3</em>: Visualization of oversampling the hard examples compared to the normal training strategy</span>
+</p>
+
+* Oversampling hard examples reduced the WER of those examples without affecting the overall WER results.
+
```
Comment (AbdulrhmnGhanem) [inline experiments/08A.md:161]: Please add the image from the comment and the total WER here.

**Review by abdoei — APPROVED** [review]

**Pull request merged**